generation/src/proof_generation/aml/syntax.py

```diff
@@ -438,8 +470,11 @@ def evar_is_fresh_ignoring_metavars(self, name: int, ignored_metavars: frozenset
     def metavars(self) -> set[MetaVar]:
         return self.pattern.metavars().union(self.plug.metavars())
 
-    def occurring_vars(self) -> set[EVar | SVar]:
-        return self.pattern.occurring_vars().difference({self.var}).union(self.plug.occurring_vars())
+    def occurring_free_vars(self) -> set[EVar | SVar]:
+        return self.pattern.occurring_free_vars().difference({self.var}).union(self.plug.occurring_free_vars())
+
+    def occurring_binder_vars(self) -> set[EVar | SVar]:
+        return self.pattern.occurring_binder_vars().difference({self.var}).union(self.plug.occurring_binder_vars())
```

Comment:
```suggestion
        return self.pattern.occurring_binder_vars().union(self.plug.occurring_binder_vars())
```

The substitution only removes free occurrences of `self.var`, so we actually have to keep it because it might still be in some binder in either `pattern` or `plug`